core/errors/index.js

```diff
@@ -0,0 +1,11 @@
+const errors = {};
+
+errors.ApplicationError = require('./application-error');
```

Comment:
Can you please add the comments on how each error are different from each other?
This will give more clarity on something like how is `ConflictError` different from `NotFoundError`

Reply:
They mostly differ by HTTP code like,

Conflict - 409
Validation - 422 
etc 

Refer: https://www.restapitutorial.com/httpstatuscodes.html

Reply:
Can you please add the comment as when to use which

Reply:
@sumitra19jha  @palisadoes  

1. Validation Error 

Suppose you are validating some things like :
- Check if the file Type is in 'png' if not throw Validation Error 
- Validation error has the capability to throw multiple validation errors at once with 422 HTTP Code

2. Conflict Error 

 - Suppose an entry is already present in DB which ideally should be immutable
 - eg: User of id 1, New user should not be added again with id 1 here it should raise Conflict error with 409 HTTP Code

3. NotFound Error 

- Anything entry which you expect to exist but not found here you will throw NotFoundError

4. UnAuthorizedError 

- Who has authentication but is Forbidden to access a resource here we will send this with 403 HTTP Code

5. UnAuthenticatedError 

- Invalid Credentials, invalid auth token 
- Here we will send UnAuthenticatedError with 401 HTTP Code

6. ApplicationError 

- ApplicationError is like interface here.

7.  InternalServerError 

- Should be thrown if we don't know the type of the error which is a very rare use case 
- eg: cant start server, db connection error. 

These are all the errors added :

errors.ApplicationError = require('./application-error');
errors.InternalServerError = require('./internal-server-error');
errors.NotFoundError = require('./not-found-error');
errors.UnauthorizedError = require('./unauthorized-error');
errors.ValidationError = require('./validation-error');
errors.UnauthenticatedError = require('./unauthenticated-error');
errors.ConflictError = require('./conflict-error');